src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/GenericContext.java

```diff
@@ -15,9 +15,34 @@
 
 package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core;
 
-public abstract class GenericContext {
-  // TODO
-  // The GenericContext class currently has only one derived class ResourceContext.
-  // We are expecting to add more different types of context and some abstract methods can be added
-  // here
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Resources;
+
+public class GenericContext {
+    protected Resources.State state;
+
+    public GenericContext(Resources.State state) {
+        this.state = state;
+    }
+
+    public Resources.State getState() {
```

Comment:
If we have `getState()`, then should we just make `state` private ?

Reply:
Done